borg/testsuite/archive.py

```diff
@@ -112,3 +112,29 @@ def test_corrupt_chunk(self):
         input = [(False, chunks[:3]), (True, [b'gar', b'bage'] + chunks[3:])]
         result = self.process(input)
         self.assert_equal(result, [{b'path': b'foo'}, {b'path': b'boo'}, {b'path': b'baz'}])
+
+
+def test_valid_msgpacked_item():
```

Comment:
since this is a top-level test function and not part of the self-tests you can use py.test features

- item_keys_serialized => fixture
- split into three tests: invalid ones (parametrize to supply data), valid ones and the third test case with the key length test